WooCommerce/Classes/Blaze/BlazeBanner.swift

```diff
@@ -3,70 +3,140 @@ import struct Yosemite.Site
 
 /// Hosting controller for `BlazeHighlightBanner`.
 final class BlazeBannerHostingController: UIHostingController<BlazeBanner> {
+    private let site: Site
+    private let containerViewController: UIViewController
+    private let dismissHandler: () -> Void
+
     init(site: Site,
-         entryPoint: EntryPoint,
-         parentViewController: UIViewController) {
-        super.init(rootView: BlazeBanner())
+         entryPoint: BlazeBanner.EntryPoint,
+         containerViewController: UIViewController,
+         analytics: Analytics = ServiceLocator.analytics,
+         dismissHandler: @escaping () -> Void) {
+        self.site = site
+        self.containerViewController = containerViewController
+        self.dismissHandler = dismissHandler
+        super.init(rootView: BlazeBanner(showsTopDivider: entryPoint.shouldShowTopDivider,
+                                         showsBottomSpacer: entryPoint.shouldShowBottomSpacer))
+
+        let blazeSource = entryPoint.blazeSource
+        analytics.track(event: .Blaze.blazeEntryPointDisplayed(source: blazeSource))
+
+        rootView.onTryBlaze = { [weak self] in
+            guard let self else { return }
+            analytics.track(event: .Blaze.blazeEntryPointTapped(source: blazeSource))
+            self.showBlaze(source: blazeSource,
+                           onCampaignCreated: dismissHandler)
```

Comment:
❓ when testing in JPiOS, creating a campaign doesn't dismiss the Blaze banner on the home screen. if users don't want to see it, I think they can just click the dismiss button easily. there might be some users who want to create another campaign after the first try.

also, "hiding" (the equivalent dismissal action) the banner in JPiOS just applies to the selected store. when switching to another store, the Blaze banner is still shown which I think makes sense - the user might not find values promoting in one store but it can be useful for a different store. this is probably an edge case though.

Reply:
> ❓ when testing in JPiOS, creating a campaign doesn't dismiss the Blaze banner on the home screen. if users don't want to see it, I think they can just click the dismiss button easily. there might be some users who want to create another campaign after the first try.

Good point - I forward this question to Ann in the design P2 post: {pecCkj-MD}-p2#comment-397. I will update the behavior in a subsequent PR if needed after getting a confirmation.

> also, "hiding" (the equivalent dismissal action) the banner in JPiOS just applies to the selected store. when switching to another store, the Blaze banner is still shown which I think makes sense - the user might not find values promoting in one store but it can be useful for a different store. this is probably an edge case though.

I've updated the user defaults to use a dictionary with the site ID as the key in 295a720f19284d28e63dae3eadd0be91f5e0b629 and 26bbf0e8368689d07c4d0b8de9e3231206c5e9f3.